app/templates/user/main-user-view.jade

```diff
@@ -12,53 +12,117 @@ block append content
   if view.user
     - var playerLevel = view.user.level()
     - var emails = view.user.getEnabledEmails()
+    - var totalGems = view.user.gems()
+    - var levelsComplete = (view.user.get('stats') || {}).gamesCompleted
+    - var heroes = view.user.heroes().length
+    - var heroID = (view.user.get('heroConfig') || {}).thangType
+    - var heroPose = view.user.getHeroPoseImage()
```

Comment:
This is an async method, so we don't want the template to be calling it–we can get it from the view if the view has it.
```suggestion
    - var heroPose = view.heroPoseImage
```

Reply:
Instead, in [the `onLoaded` method of `MainUserView.coffee`](https://github.com/codecombat/codecombat/blob/master/app/views/user/MainUserView.coffee#L61), add some code like this:
```coffee
    loadHeroPoseImage = =>
      @heroPoseImage = @user.getHeroPoseImage()
      @render()
    _.defer @loadHeroPoseImage
```

Reply:
I get a 
`lodash.js?e986:5493 Uncaught TypeError
    at Function.defer (lodash.js?e986:5493)
    at MainUserView.m`
error when I run defer. It looks like lodash.js does exist though.

Reply:
Weird. I have
```
getHeroPoseImage: co.wrap ->
    options={}
    if ((@get('heroConfig') or {}).thangType)
      poseFile = yield fetchJson("/db/thang.type/#{((@get('heroConfig') or {}).thangType)}/version",options)
      return '/file/' + poseFile.poseImage
```
in Users.coffee
```
loadHeroPoseImage: =>
    @heroPoseImage = @user.getHeroPoseImage
    console.log(@heroPoseImage)
    @heroPoseImage.then(result) ->
      @heroPoseImage = result
      @render()

  onLoaded: ->
    _.defer @loadHeroPoseImage
```
In MainUserView
```
- var heroPose = view.heroPoseImage
```
But it's getting me
```
createPromise() {
    return co.call(this, fn.apply(this, arguments));
  }
``` 
instead of the link :(

Reply:
Oh I fixed it.